MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -10,29 +10,77 @@ import Foundation
 
 
 class QuestionFactory: QuestionFactoryProtocol {
+    private let moviesLoader: MoviesLoading
     weak var delegate: QuestionFactoryDelegate?
-    init(delegate: QuestionFactoryDelegate) {
+    init(moviesLoader: MoviesLoading, delegate: QuestionFactoryDelegate?) {
+        self.moviesLoader = moviesLoader
         self.delegate = delegate
     }
-    private let questions: [QuizQuestion] = [
-        QuizQuestion(image: "The Godfather", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "The Dark Knight", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "Kill Bill", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "The Avengers", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "Deadpool", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "The Green Knight", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
-        QuizQuestion(image: "Old", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
-        QuizQuestion(image: "The Ice Age Adventures of Buck Wild", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
-        QuizQuestion(image: "Tesla", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
-        QuizQuestion(image: "Vivarium", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false)
-    ]
+    func loadData() {
+        moviesLoader.loadMovies { [weak self] result in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                switch result {
+                case .success(let mostPopularMovies):
+                    self.movies = mostPopularMovies.items
+                    self.delegate?.didLoadDataFromServer()
+                case .failure(let error):
+                    DispatchQueue.main.async { [weak self] in
+                        guard let self else { return }
+                        self.delegate?.didFailToLoadData(with: error)
+                    }
```

Comment:
Можно не разыменовывать, а писать через ?, так получится короче: 
```suggestion
                    DispatchQueue.main.async { [weak self] in
                        self?.delegate?.didFailToLoadData(with: error)
                    }
```
таким вариантом можно пользоваться, когда в замыкании всего одна-две строчки. 

Также замчечу, что блок switch result уже помещен в  DispatchQueue.main.async , поэтому и этот участок будет выполнен на главной очереди. Второй DispatchQueue.main.async  - избыточен: 
```suggestion
                    self.delegate?.didFailToLoadData(with: error)
```

Reply:
Это нужно поправить к следующему ревью